pkg/acl/access_control_list_test.go

```diff
@@ -34,13 +34,97 @@ func Test_DeniedUsers_IsUserDenied(t *testing.T) {
 			want:           true,
 		},
 	}
+	RegisterTestingT(t)
 	for _, tt := range tests {
 		tt := tt
 		t.Run(tt.name, func(t *testing.T) {
 			t.Parallel()
-			RegisterTestingT(t)
 			ok := tt.deniedAccounts.IsUserDenied(tt.arg)
 			Expect(ok).To(Equal(tt.want))
 		})
 	}
 }
+
+func Test_ReadDenyListConfigFile(t *testing.T) {
+	type args struct {
+		file        string
+		deniedUsers *DeniedUsers
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "should return an error when attempting to read deny list with non-existent file",
+			args: args{
+				file:        "non-existent path",
+				deniedUsers: &DeniedUsers{},
+			},
+			wantErr: true,
+		},
+		{
+			name: "should not return an error when attempting to read deny list with non-existent file",
+			args: args{
+				file:        "config/deny-list-configuration.yaml",
+				deniedUsers: &DeniedUsers{},
+			},
+			wantErr: false,
+		},
+	}
+
+	RegisterTestingT(t)
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := readDenyListConfigFile(tt.args.file, tt.args.deniedUsers)
+			Expect(err != nil).To(Equal(tt.wantErr))
+		})
+	}
+}
+
+func Test_ReadFiles(t *testing.T) {
+	type args struct {
+		file           string
+		deniedUsers    *DeniedUsers
+		enableDenyList bool
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "should not return an error when deny list is enabled and config file path is valid",
+			args: args{
+				file:           "config/deny-list-configuration.yaml",
+				deniedUsers:    &DeniedUsers{},
+				enableDenyList: true,
+			},
+			wantErr: false,
+		},
+		{
+			name: "should not return an error when deny list is disabled",
+			args: args{
+				file:           "config/deny-list-configuration.yaml",
+				deniedUsers:    &DeniedUsers{},
+				enableDenyList: false,
+			},
+			wantErr: false,
+		},
```

Comment:
There are missing test cases for when file path is not valid with deny list enabled/disabled?

Reply:
good point. added tests to address both comments

Reply:
Thanks!